agent/agent.go

```diff
@@ -156,6 +156,10 @@ LOOP:
 				grip.Notice("Next task response indicates agent should exit")
 				return nil
 			}
+			// if the host's current task group is finished we teardown
+			if nextTask.ShouldTeardownGroup {
+				needPostGroup = nextTask.ShouldTeardownGroup
```

Comment:
Instead of setting `needPostGroup` here, could you eliminate that variable entirely by
```
if nextTask.ShouldTeardownGroup {
// things from else if needPostGroup section
continue LOOP
```

Reply:
I think you might be able to get rid of it everywhere.

Reply:
Hmm I want to be careful about that assumption, `ShouldTeardownGroup` is only set if there _was_ a task group and now it's a different task group, whereas needPostGroup basically runs whenever there's no task? Is that not something we might need to consider?

Reply:
Ah, I see what you mean. That makes sense.